Utilities/Hash.cpp

```diff
@@ -0,0 +1,52 @@
+#include "stdafx.h"
+#include "Hash.h"
+#include <sstream>
+#include <iomanip>
+
+#include "../rpcs3/Crypto/sha1.h"
+
+namespace hash
+{
+
+std::string SHA1(const std::string& str)
+{
+	int len = str.size();
+
+	unsigned char hash[20];
+	const unsigned char* input = [&]
+	{
+		unsigned char* bytes = new unsigned char[len + 1];
```

Comment:
I had some problems converting ```std::string``` to ```unsigned char*``` for the  ```sha1``` function. I noticed the memory increase because of the missing free. Now I switched over to ```std::unique_ptr``` and this is much better. I spammed this function in loop 1000000 times and the memory didn't increased this time. :)